src/domain/UBGraphicsScene.cpp

```diff
@@ -302,18 +306,9 @@ void UBZLayerController::shiftStoredZValue(QGraphicsItem *item, qreal zValue)
 /**
  * @brief Returns true if the zLevel is not used by any item on the scene, or false if so.
  */
-bool UBZLayerController::zLevelAvailable(QGraphicsItem* item)
+bool UBZLayerController::zLevelAvailable(const qreal zLevel) const
 {
-    foreach(QGraphicsItem* it, mScene->items())
-    {
-        if (item != it)
-        {
-            if (it->zValue() == item->zValue())
-                return false;
-        }
-    }
-
-    return true;
+    return mAlreadyUsedZLevels.contains(zLevel);
```

Comment:
This should 

`return !mAlreadyUsedZLevels.contains(zLevel);`

The level is available if it is **not** contained in the set.